app/controllers/authentication.js

```diff
@@ -115,167 +116,188 @@ var Controller = function(){
-              return res.send({ token: token });
-            }
-            var user = new User();
-            user.github = profile.id;
-            user.picture = profile.avatar_url;
-            user.displayName = profile.name;
-            user.email = profile.email;
-            user.save(function(error) {
-              console.log(error);
-              var token = createJWT(user);
-              res.send({ token: token });
-            });
+            message: 'Did not get email address.'
           });
         }
-      });
-    });
-  }
 
-    /*
-   |--------------------------------------------------------------------------
-   | Login with Google
-   |--------------------------------------------------------------------------
-   */
-  this.google = function(req, res) {
-    var accessTokenUrl = 'https://accounts.google.com/o/oauth2/token';
-    var peopleApiUrl = 'https://www.googleapis.com/plus/v1/people/me/openIdConnect';
-    var params = {
-      code: req.body.code,
-      client_id: req.body.clientId,
-      client_secret: config.GOOGLE_SECRET,
-      redirect_uri: req.body.redirectUri,
-      grant_type: 'authorization_code'
-    };
+        // Retrieve organization authorization information for the user.
+        request.get({ url: organizationUrl + profile.login,
+                      qs: accessToken,
+                      headers: headers,
+                      json: true
+                    },
+                    function(err, response)
+                    {
+                      // Test if the user belongs to the organization(User authorization).
+                      if (response.statusCode != 200)
+                      {
+                        return res.status(401).send({
+                          message: 'You do not have required access.'
+                        })
+                      }
 
-    // Step 1. Exchange authorization code for access token.
-    request.post(accessTokenUrl, { json: true, form: params }, function(err, response, token) {
-      var accessToken = token.access_token;
-      var headers = { Authorization: 'Bearer ' + accessToken };
+                      // Retrieve admin authorization information for the user.
+                      request.get({ url: teamAdminUrl + profile.login,
+                                    qs: accessToken,
+                                    headers: headers,
+                                    json: true
+                                  },
+                                  function(err, response)
+                                  {
+                                    User.findOne({github: profile.id}, function(err, existingUser)
+                                    {
+                                      // Check if the user account exists.
+                                      if (existingUser)
+                                      {
+                                        console.log("Return an existing user account.");
```

Comment:
please could you replace these console.logs with `winston` -logger (it is used elsewhere as well)